R/reorder.R

```diff
@@ -45,15 +49,24 @@
 #'   geom_point() +
 #'   geom_line() +
 #'   labs(colour = "Chick")
-fct_reorder <- function(.f, .x, .fun = median, ..., .desc = FALSE) {
+fct_reorder <- function(.f, .x, .fun = median, ..., .na_rm = FALSE, .na_last = TRUE, .desc = FALSE) {
   f <- check_factor(.f)
   stopifnot(length(f) == length(.x))
   check_dots_used()
 
+  miss <- is.na(.x)
+  if (any(miss) && !isTRUE(.na_rm)) {
+    cli::cli_warn(c(
+      "{.fn fct_reorder} removing {sum(miss)} missing value{?s}.",
+      i = "Use `.na_rm = TRUE` to silence this message"
+    ))
+  }
```

Comment:
I'm slightly uncomfortable with the fact that you can't _not remove_ missing values, i.e. if you expect your summary function to handle them somehow. What about:

- `.na_rm = NULL`. Does `TRUE` with a warning if there are `NA`s in `.x`. This is the default. Warning encourages explicitly setting `.na_rm`.
- `.na_rm = FALSE`. Does nothing.
- `.na_rm = TRUE`. Does `TRUE` silently.